draft-ietf-tls-external-psk-importer.md

```diff
@@ -99,53 +101,68 @@ a tuple of (Base Key, External Identity, KDF). The associated KDF (and hash func
 - External Identity: The identity of an EPSK.
 - Imported Identity: The identity of a PSK as sent on the wire.
 
+## Notation and Cryptographic Dependencies
+
+The key importer API depends on a generic KDF which provides the following interface.
+
+- Nh: The output size of the Extract function.
+- Extract(salt, IKM): Extract a pseudorandom key of fixed length Nh
+  from input keying material `IKM` and an optional octet string
+  `salt`.
+- Expand(PRK, info, L): Expand a pseudorandom key `PRK` using
+  optional string `info` into `L` bytes of output keying material.
+
+It also depends on the following utility function.
+
+- `concat(x0, ..., xN)`: Concatenation of octet strings.
+  `concat(0x01, 0x0203, 0x040506) = 0x010203040506`
+
 # Key Import
 
 A key importer takes as input an EPSK with external identity 'external_identity' and base key 'epsk',
-as defined in {{terminology}}, along with an optional label, and transforms
-it into a set of PSKs and imported identities for use in a connection based on supported HashAlgorithms.
-In particular, for each supported HashAlgorithm 'hash', the importer constructs an ImportedIdentity
-structure as follows:
+as defined in {{terminology}}, along with an optional context string, and transforms
+it into a set of PSKs and imported identities for use in a connection based on the (D)TLS protocol
+version and a specific KDF. In particular, for each supported protocol version 'protocol' and KDF
+function 'kdf', the importer constructs an ImportedIdentity structure as follows:
```

Comment:
I think that it might make sense to use the phrase "target protocol" and "target KDF" or something similar here.  There's a risk of having this confused in some way with the KDF that is being applied.

Reply:
Good suggestion! I'll work that in somehow.